packages/flutter_tools/test/src/cache_test.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+import 'package:platform/platform.dart';
+
+import 'package:flutter_tools/src/cache.dart';
+
+import 'context.dart';
+
+void main() {
+  group('$Cache.checkLockAcquired', () {
+    setUp(() {
+      Cache.enableLocking();
+    });
+
+    tearDown(() {
+      Cache.enableLocking();
```

Comment:
This is intentional. I want the tests to begin and end with enabled locking. The former for all tests in this group. The latter to prevent side-effects outside it. But the fact you had to ask tells me that this is not obvious. So I left a comment.